### PR TITLE
add GraphQL-DQL section

### PR DESCRIPTION
**Pull request by rderbier**
Explaining how DQL and GraphQL can be used together.

please review all pages in /graphql-dql

### CODE REVIEW

**Comment by rderbier** [comment]
Some pages are just pointers, we can add more info later.

File: content/graphql-dql/graphql-dql-schema.md [hunk 1]
```diff
@@ -0,0 +1,265 @@
++++
+title = "GraphQL and DQL schemas"
+weight = 1
+[menu.main]
+  name = "GraphQL and DQL schemas"
+  identifier = "graphql-dql-schema"
+  parent = "graphql-dql"
++++
+
+The first step in mastering DQL in the context of GraphQL API is probably to understand the fundamental difference between GraphQL schema and DQL schema.
```
Comment (matthewmcneely) [inline content/graphql-dql/graphql-dql-schema.md:10]: strike 'probably'

File: content/graphql-dql/graphql-dql-schema.md [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+type Review {
+    id: ID!
+    about: Product!
+    by: Customer!
+    comment: String @search(by: [fulltext])
+    rating: Int @search
+}
+```
+
+The API and the engine logic are generated from the schema defining the types of objects we are dealing with, the fields, and the relationships in the form of fields referencing other types.
```
Comment (matthewmcneely) [inline content/graphql-dql/graphql-dql-schema.md:38]: This sentence tripped me up. I asked GPT to rework for clarity:

We generate the API and engine logic based on a schema that defines the types of objects we're working with, as well as their fields and relationships, which are expressed as fields that reference other types.

File: content/graphql-dql/graphql-dql-schema.md [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+
+On the Dgraph custer which has a [schema mode]({{<relref "schema-modes">}}) set to ``flexible`` (which is the default behavior) you can run a mutation without declaring any schema.
+
+For example, you can run the following mutation:
+```graphql
+{
+  set {
+    <_:jedi1> <character_name> "Luke Skywalker" .
+    <_:leia> <character_name> "Leia" .
+    <_:sith1> <character_name> "Anakin" (aka="Darth Vador",villain=true).
```
Comment (matthewmcneely) [inline content/graphql-dql/graphql-dql-schema.md:51]: Should there be a space before the closing dot?

File: content/graphql-dql/graphql-dql-schema.md [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+    <_:sith1> <has_for_child> <_:jedi1> .
+    <_:sith1> <has_for_child> <_:leia> .
+  } 
+}
+```
+This mutation using the [RDF]({{< relref "dql-rdf">}}) notation, could be read as:
+
+- There is a first `thing`, that we refer to as `jedi1`, having a `character name` “Luke Skywalker”, 
+- There is anothe `thing`, that we refer to as `leia`, having a `character name` “Leia”,
```
Comment (matthewmcneely) [inline content/graphql-dql/graphql-dql-schema.md:60]: typo

File: content/graphql-dql/graphql-dql-schema.md [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+```
+type Person {
+   Person.name
+   Person.friends
+}
+```
+
+Once again, the DQL type is just a declaration of the list of predicates that one can expect to be present in a node of having ``dgraph.type`` equal ``Person``.
+
+The default mapping can be customized by using the [@dgraph directive]({{< relref "directive-dgraph" >}}).
```
Comment (matthewmcneely) [inline content/graphql-dql/graphql-dql-schema.md:173]: Worth linking to this directive?

File: content/graphql/schema/directives/directive-dgraph.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+[menu.main]
+    parent = "directives"
+    identifier = "directive-dgraph"
++++
+
+
+The `@dgraph` directive customizes the name of the types and predicates generated in Dgraph when deploying a GraphQL Schema.
+
+* `type <type> @dgraph(type: "TypeNameToUseInDgraph")` controls what Dgraph type is used for a GraphQL type.
+* `field: SomeType @dgraph(pred: "DgraphPredicate")` controls what Dgraph predicate is mapped to a GraphQL field.
```
Comment (matthewmcneely) [inline content/graphql/schema/directives/directive-dgraph.md:13]: Following the convention you introduced, shouldn't SomeType be `<SomeType>`?

File: content/graphql-dql/graphql-data-loading.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
++++
+
+
+
+After you have deployed your first GraphQL Schema, you get a GraphQL API served on ``/graphql`` endpoint and an empty backend. You can populate the graph database using the mutations operations on the GraphQL API.
+
+A more efficient way to populate the database is to use the Dgraph's [import tools]({{< relref "importdata">}}).
+
+The first step is to understand the [schema mapping]({{<relref "graphql-dql-schema">}}) and to prepare your RDF files or JSON files to follow the internal Dgraph predicates names. 
+You also have to make sure that you properly generate data for the `dgraph.type` predicate so that each node is asscociated with it's type.
```
Comment (matthewmcneely) [inline content/graphql-dql/graphql-data-loading.md:17]: it's should be its

File: content/graphql-dql/graphql-data-loading.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+If you are using the [initial import]({{<relref "bulk-loader">}}) tool, you can provide the GraphQL schema along with the data to import when executing the bulk load.
```
Comment (matthewmcneely) [inline content/graphql-dql/graphql-data-loading.md:19]: You also have to supply a DQL schema (in fact that's the only requirement). I'm not even sure why there's a graphql schema flag.

One way to retrieve the DQL schema:
`curl localhost:8080/query -d 'schema {}' --header 'content-type: application/dql' | jq '.data.schema' > schema/schema.dql`

But maybe this belongs in the bulk loader docs.

File: content/graphql-dql/graphql-data-migration.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+  parent = "graphql-dql"
++++
+
+
+
+When deploying a new version of your GraphQL Schema, Dgraph will update the underlying DQL Schema but will not alter the data.
+
+As explained in [GraphQL and DQL Schemas]({{<relref "graphql-dql-schema">}}) overview, Dgraph has no constraints at the database level and any node with predicates is valid. 
+
+You may face with several data GraphQL API and data discrepancies.
```
Comment (matthewmcneely) [inline content/graphql-dql/graphql-data-migration.md:16]: 'You may be faced' or 'You may face several' ?

File: content/graphql-dql/graphql-data-migration.md [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+
+Then you update the Schema and remove the field.
+```graphql
+type TestDataMigration {
+  id: ID!
+  someInfo: String!
+}
+```
+
+The data you have previously created is still in the graph database !
```
Comment (matthewmcneely) [inline content/graphql-dql/graphql-data-migration.md:38]: No need for the exclamation. It make it seem like it's a mistake.

File: content/graphql-dql/graphql-data-migration.md [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+
+Moreover if you delete the `TestDataMigration` object using its `id`, the GraphQL API delete operation will be successful.
+
+If you followed the [GraphQL - DQL Schema mapping]({{<relref "graphql-dql-schema">}}), you understand that Dgraph has used the list the known list of predicates (id, someInfo) and removed them. In fact, Dgraph also removed the `dgraph.type` predicate and so this `TestDataMigration` node is not visible anymore to the GraphQL API.
+
+The point is that a node with this `uid` exists and has a predicate `someOtherInfo`. This is because this data has been created initialy and nothing in the process of deploying a new version and then using a delete operation by ID instructed Dgraph to delete this predicate.
+
+You end up with a node without type (i.e without a `dgraph.type` predicate) and with an old predicate value which is 'invisible' to your GraphQL API!
```
Comment (matthewmcneely) [inline content/graphql-dql/graphql-data-migration.md:46]: Again, the exclamation point is unnecessary

File: content/graphql-dql/graphql-data-migration.md [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+
+When doing a GraphQL schema deployement, you must take care of the data cleaning and data migration.
+The good news is that DQL offers you the tools to identify (search) potential issues and to correct the data (mutations).
+
+In the previous case, you can alter the database and completely delete the predicate or you can write an 'upsert' DQL query that will search the nodes of interest and delete the unused predicate for those nodes.
+
+### new non-nullable field
+Another obvious example appears if you deploy a new version containing a new non-nullable field for an existing type. The existing 'nodes' of the same type in the graph do not have this predicate. A Gra[hQL query reaching those nodes will return a list of errors. You can easily write an 'upsert' DQL mutation to find all node of this type not having the new predicate and update them with a default value.
```
Comment (matthewmcneely) [inline content/graphql-dql/graphql-data-migration.md:54]: Gra[hQL

**Pull request merged**